solutions/platinum/cses-2417.mdx

```diff
@@ -0,0 +1,100 @@
+---
+id: cses-2417
+source: CSES
+title: Counting Coprime Pairs
+author: Alexis Tao
+---
+
+## Implementation
+
+**Time Complexity:** $\mathcal O(N \cdot \sqrt{\max (x_i)})$
+
+<LanguageSection>
+
+<CPPSection>
+
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+const int MAX_X = 1e6 + 1;
+
+int main() {
+	ll ans = 0;
+	int N;
+	vector<int> dp(MAX_X, 0);
+	vector<bool> is_prime(MAX_X, true);
+	vector<int> primes;
+
+	// Lambda to initialize the sieve and populate the primes vector
+	auto init = [&]() {
+		fill(is_prime.begin() + 2, is_prime.end(),
+		     true);  // Mark all numbers as prime initially (except 0 and 1)
```

Comment:
formatting looks kind of weird here, why don't you just have the comment on a separate line?

Reply:
this line is also unnecessary? you only call the function once, just set is_prime[0]=is_prime[1]=false